src/zimscraperlib/zim/creator.py

```diff
@@ -146,7 +150,31 @@ def config_indexing(
         self.__indexing_configured = True
         return self
 
+    def _maybe_metadata_from_maybe_illustration(self, name, value):
+        if not isinstance(name, str) or not name.startswith("Illustration_"):
+            return None
+        if isinstance(value, bytes):
+            image = io.BytesIO(value)
+        try:
+            img = PIL.Image.open(image)
+            return img.format
+        except Exception:
+            return None
+
+    def _log_metadata(self):
+        if logger.isEnabledFor(DEBUG):
+            for name, value in sorted(self._metadata.items()):
+                illustration_metadata = self._maybe_metadata_from_maybe_illustration(
+                    name, value
+                )
+                if illustration_metadata is not None:
+                    logger.debug(f"Metadata: {name} MD = {illustration_metadata}")
+                else:
+                    logger.debug(f"Metadata: {name} = {value}")
```

Comment:
What if `value` is a `bytes` because parsing image format failed?

Reply:
Min-viable is to simply log value in full, even if it's huge. Doesn't seem to come up in the tests I've looked at.

Reply:
Simple enough to output something like "Failed to parse, 18043 bytes"